front/src/common/Map/WarpedMap/SimulationWarpedMap.tsx

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable no-console */
+import { useSelector } from 'react-redux';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { clamp, first, isEmpty, isNil, keyBy, last, mapValues, omitBy } from 'lodash';
+import { PiLinkBold, PiLinkBreakBold } from 'react-icons/pi';
+
+import bbox from '@turf/bbox';
+import length from '@turf/length';
+import { lineString } from '@turf/helpers';
+import { BBox2d } from '@turf/helpers/dist/js/lib/geojson';
+import { Feature, FeatureCollection, LineString, Position } from 'geojson';
+import { LngLatBoundsLike } from 'maplibre-gl';
+
+import { RootState } from 'reducers';
+import { LoaderFill } from 'common/Loader';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { LayerType } from 'applications/editor/tools/types';
+import DataLoader from 'common/Map/WarpedMap/DataLoader';
+import getWarping, { WarpingFunction } from 'common/Map/WarpedMap/getWarping';
+import WarpedMap from 'common/Map/WarpedMap/WarpedMap';
+import { TrainPosition } from 'applications/operationalStudies/components/SimulationResults/SimulationResultsMap/types';
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { getImprovedOSRDData } from 'common/Map/WarpedMap/core/helpers';
+import { getSelectedTrain } from 'reducers/osrdsimulation/selectors';
+import { AsyncMemoState, getAsyncMemoData, useAsyncMemo } from 'utils/useAsyncMemo';
+import { getSimulationHoverPositions } from 'applications/operationalStudies/components/SimulationResults/SimulationResultsMap/helpers';
+import { clip } from 'utils/mapHelper';
+
+import './SimulationWarpedMap.scss';
+import { Train } from '../../../reducers/osrdsimulation/types';
+
+const TIME_LABEL = 'Warping OSRD and OSM data';
+const WIDTH = 300;
+
+interface PathStatePayload {
+  path: Feature<LineString>;
+  pathBBox: BBox2d;
+  regularBBox: BBox2d;
+  transform: WarpingFunction;
+}
+
+interface DataStatePayload {
+  osm: Record<string, FeatureCollection>;
+  osrd: Partial<Record<LayerType, FeatureCollection>>;
+}
+
+/**
+ * This component handles loading the simulation path, all the surrounding data (OSM and OSRD), transforms them, and
+ * then mounts a WarpedMap with all that data:
+ */
+const SimulationWarpedMap: FC<{ collapsed?: boolean }> = ({ collapsed }) => {
+  const infraID = useSelector(getInfraID);
+  const [state, setState] = useState<
+    | { type: 'idle' }
+    | { type: 'loading' }
+    | { type: 'error'; message?: string }
+    | ({
+        type: 'pathLoaded';
+      } & PathStatePayload)
+    | ({
+        type: 'dataLoaded';
+      } & PathStatePayload &
+        DataStatePayload)
+  >({ type: 'idle' });
+  const pathfindingID = useSelector(
+    (s: RootState) => s.osrdsimulation.selectedProjection?.path
+  ) as number;
+  const [getPath] = osrdEditoastApi.useLazyGetPathfindingByIdQuery();
```

Comment:
It's better to use rtk definition like this, so we can more easily go to definition and see the types
```suggestion
  const [getPath] = osrdEditoastApi.endpoints.getPathfindingById.useLazyQuery();
```